src/middlewared/middlewared/service.py

```diff
@@ -399,6 +399,13 @@ def get_jobs(self, filters=None, options=None):
         ], filters, options)
         return jobs
 
+    @accepts(Int('id'))
+    @job()
+    def job_wait(self, job, id):
```

Comment:
What I meant here earlier is that on error we should probably raise it, otherwise return the result? Thoughts?

Reply:
I don’t think that is a good idea. The caller will know if the job failed or not via the return value and it can work it out accordingly. Thoughts ?

Reply:
I think explicit failure is always better than implicit

Reply:
So if another job is waiting for for a different job, i don’t think it is a good idea to let the exception raise this way. Thoughts @skarekrow @william-gr  ?

Reply:
I think the caller should be expecting and catching in an exception in that case. Silently discarding them or relying on return values seems odd to me.

Reply:
We are not discarding ;) everything is available when we query for the job :p But please let me know what would be best and i can move ahead then

Reply:
Idea is we wait for the job to complete. Not interpret how it ends, that’s my argument

Reply:
But you're not getting the Job object, just an error attribute which doesnt tell much.

So it either returns the whole job or we follow the standard practice of raising an error if something bad happened or the final result otherwise.